### PR TITLE
local info: refactor information about the local env

### PR DESCRIPTION
**Pull request by mattklein123**
This has been bothering me for a while. We are copying things like local
zone, local cluster, etc. all over the place. It's also likely that we will
add more things in the future. This introduces a LocalInfo abstract class
which holds such information.

The change is large by number of lines but it's simple.

### CODE REVIEW

**Comment by mattklein123** [comment]
@lyft/network-team

File: include/envoy/server/options.h [hunk 1]
```diff
@@ -51,21 +51,6 @@ class Options {
   virtual uint64_t restartEpoch() PURE;
 
   /**
-   * @return const std::string& the service cluster name where the server is running.
```
Comment (wgallagher) [inline include/envoy/server/options.h]: move these comments to LocalInfo

Comment (mattklein123) [inline include/envoy/server/options.h]: already fixed.

File: include/envoy/local_info/local_info.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Information about the local environment.
+ */
+class LocalInfo {
+public:
+  virtual ~LocalInfo() {}
+
+  /**
+   * Human readable network address.
```
Comment (RomanDzhabarov) [inline include/envoy/local_info/local_info.h]: nit: can you put also example for this. Human readable network address, e.g., ...

**Comment by RomanDzhabarov** [comment]
+1

**Pull request merged**